### PR TITLE
KAFKA-8093: fixed some javadoc errors

### PR DESCRIPTION
**Pull request by pkleindl**
corrected link to Oracle documentation according to https://discuss.gradle.org/t/javadoc-task-link-not-linking-external-javadocs/25264/3
Question: Should the link really go to 7?

If I run "aggregatedJavadoc" locally I get ERRORs because of
exclude "**/internals/**":

> Task :aggregatedJavadoc
/Users/patrik.kleindl/IdeaProjects/kafka_pkleindl/streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java:31: error: cannot find symbol
import org.apache.kafka.streams.internals.QuietStreamsConfig;
                                         ^
  symbol:   class QuietStreamsConfig
  location: package org.apache.kafka.streams.internals
/Users/patrik.kleindl/IdeaProjects/kafka_pkleindl/streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java:43: error: cannot find symbol
import org.apache.kafka.streams.internals.KeyValueStoreFacade;
                                         ^
  symbol:   class KeyValueStoreFacade
  location: package org.apache.kafka.streams.internals
/Users/patrik.kleindl/IdeaProjects/kafka_pkleindl/streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java:44: error: cannot find symbol
import org.apache.kafka.streams.internals.QuietStreamsConfig;
                                         ^
  symbol:   class QuietStreamsConfig
  location: package org.apache.kafka.streams.internals
/Users/patrik.kleindl/IdeaProjects/kafka_pkleindl/streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java:45: error: cannot find symbol
import org.apache.kafka.streams.internals.WindowStoreFacade;
                                         ^
  symbol:   class WindowStoreFacade
  location: package org.apache.kafka.streams.internals
4 errors

I guess the problem is that this excludes everything in the path of internals (including test-utils), maybe this was only meant for stuff inside streams?

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by vvcephei — APPROVED** [review]
LGTM. Thanks, @pkleindl !

I'm not sure what to do about the problem with the exclusions. I'm guessing we didn't want to generate the docs for the internals classes, which I'd agree with.

File: build.gradle [hunk 1]
```diff
@@ -1418,7 +1418,7 @@ project(':connect:api') {
 
   javadoc {
     include "**/org/apache/kafka/connect/**" // needed for the `javadocAll` task
-    options.links "http://docs.oracle.com/javase/7/docs/api/"
+    options.links "https://docs.oracle.com/javase/7/docs/api/"
```
Comment (mjsax) [inline build.gradle:1421]: @ijuma Is this correct?

**Comment by mjsax** [comment]
@pkleindl Thanks for the PR.

There is also
```
/Users/matthias/IdeaProjects/kafka/connect/api/src/main/java/org/apache/kafka/connect/header/Headers.java:261: warning - @Header is an unknown tag.
```

That is listed on the JIRA.

\cc @rhauch for review

File: build.gradle (hunk 1 above)
Comment (ijuma) [reply build.gradle:1421]: We updated this in other PRs. This needs to be rebased.

Comment (pkleindl) [reply build.gradle:1421]: @ijuma I did the rebase and tried to push after github didn't show any change, but it seems this has now included all commits here, I don't know why.

**Comment by pkleindl** [comment]
@mjsax After updating everything it seems the other issue with the Headers was fixed meanwhile by
825fa3fa Mickael Maison
If you could verify this and this PR is no longer needed and KAFKA-8093 can be closed.

**Comment by mjsax** [comment]
Ack. Thanks.

**Pull request closed without merging**